kernel-api/src/main/scala/org/apache/toree/dependencies/CoursierDependencyDownloader.scala

```diff
@@ -313,19 +313,19 @@ class CoursierDependencyDownloader extends DependencyDownloader {
    * @return The resulting URIs
    */
   private def repositoriesToURIs(repositories: Seq[Repository]) = repositories.map {
-    case IvyRepository(pattern, _, _, _, _, _, _, _, _)  => pattern
-    case MavenRepository(root, _, _, _)                  => root
+    case IvyRepository(pattern, _, _, _, _, _, _, _)  => pattern.string
```

Comment:
So i think that rather than having this pattern matching with all the underscores, we probably want to just do something like

```scala
case ivy: IvyRepository => ivy.pattern.string
```

and similar in other places.  That should allow a bit more robustness for when coursier changes again.